app/models/atmosphere/appliance.rb

```diff
@@ -129,15 +127,40 @@ def owned_by?(user)
       appliance_set.user_id == user.id
     end
 
+    def prepaid_until
+      # Helper method which determines how long this appliance will remain prepaid
+      if deployments.blank?
+        # Invoking this method on an appliance which has no deployments is a "nasal demons"
+        # scenario (the question doesn't make sense and neither will the answer).
+        # Returning fixed time is least likely to break anything on the requestor side.
+        result = Time.parse('2000-01-01 12:00')
+      else
+        result = deployments.max_by { |dep| dep.prepaid_until }.prepaid_until
+      end
+      result
+    end
+
+    # This method provided for backward compatibility
+    def billing_state
+      if deployments.any?{|d| d.billing_state == 'prepaid'}
```

Comment:
Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.